javaparser-core/src/main/java/com/github/javaparser/printer/PrintableSource.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2011, 2013-2020 The JavaParser Team.
+ *
+ * This file is part of JavaParser.
+ *
+ * JavaParser can be used either under the terms of
+ * a) the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * b) the terms of the Apache License
+ *
+ * You should have received a copy of both licenses in LICENCE.LGPL and
+ * LICENCE.APACHE. Please refer to those files for details.
+ *
+ * JavaParser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+
+package com.github.javaparser.printer;
+
+import com.github.javaparser.Position;
+
+public interface PrintableSource {
```

Comment:
Please add JavaDoc explaining what exactly a printable source is. Presumably this is a node, but without examining the rest of the files it's not immediately clear.

Reply:
i will do this asap

Reply:
I remove this interface because SourcePrinter is an implementation detail. No need for an interface and another implementation.